app/components/form/fields/TextField.tsx

```diff
@@ -56,6 +56,8 @@ export interface TextFieldProps<
   units?: string
   validate?: Validate<FieldPathValue<TFieldValues, TName>, TFieldValues>
   control: Control<TFieldValues>
+  /** Alters the value of the input during the field's onChange event. */
+  transform?: (value: string) => FieldPathValue<TFieldValues, TName>
```

Comment:
I changed the return type from `string | undefined` to this, which effectively says it returns whatever the type of the field in question is. The explicit `string | undefined` was fine for now, but we'd run into a type error later when we tried to use this on a field with a different type.